### PR TITLE
[wasm] Enable SIMD

### PR DESCRIPTION
**Pull request by radekdoulik**
Change default value of `WasmEnableSIMD` property to true. That enables SIMD for wasm/AOT.

Use docker image with newer v8 and node on helix.

### CODE REVIEW

**Comment by radekdoulik** [comment]
Currently we fail on helix because v8 is too old. I am trying to improve that with 2 independent approaches, https://github.com/dotnet/runtime/pull/80808 and https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/774

**Comment by radical** [comment]
[wbt failure]( https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-80797-merge-fd8d36aa7ab4451092/Workloads-Wasm.Build.Tests.NativeBuildTests/1/console.7e7856c0.log?helixlogtype=result):
```
Precompiling failed for /root/helix/work/workitem/e/wbt/3ca4v33t.hq2/obj/Release/net8.0/browser-wasm/wasm/for-publish/aot-in/System.Private.CoreLib.dll with exit code 134. [/root/helix/work/workitem/e/wbt/3ca4v33t.hq2/mono_aot_cross_Release_True.csproj]
Mono Ahead of Time compiler - compiling assembly /root/helix/work/workitem/e/wbt/3ca4v33t.hq2/obj/Release/net8.0/browser-wasm/wasm/for-publish/aot-in/System.Private.CoreLib.dll [/root/helix/work/workitem/e/wbt/3ca4v33t.hq2/mono_aot_cross_Release_True.csproj]
  %6 = load %System.Numerics.Vector4, %System.Numerics.Vector4* %5, align 1 [/root/helix/work/workitem/e/wbt/3ca4v33t.hq2/mono_aot_cross_Release_True.csproj]
<4 x float> [/root/helix/work/workitem/e/wbt/3ca4v33t.hq2/mono_aot_cross_Release_True.csproj]
* Assertion: should not be reached at /__w/1/s/src/mono/mono/mini/mini-llvm.c:1455 [/root/helix/work/workitem/e/wbt/3ca4v33t.hq2/mono_aot_cross_Release_True.csproj]
```

**Review by lewing — COMMENTED** [review]
LGTM but we need to resolve the wbt failure

File: src/libraries/sendtohelix-wasm.targets [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
     <NeedsWorkload Condition="'$(Scenario)' == 'BuildWasmApps'">true</NeedsWorkload>
     <NeedsEMSDK Condition="'$(NeedsToBuildWasmAppsOnHelix)' == 'true'">true</NeedsEMSDK>
-    <NeedsEMSDKNode Condition="'$(Scenario)' == 'WasmTestOnNodeJS' or '$(Scenario)' == 'BuildWasmApps'">true</NeedsEMSDKNode>
+    <NeedsEMSDKNode Condition="'$(Scenario)' == 'WasmTestOnNodeJS' or '$(Scenario)' == 'BuildWasmApps'">false</NeedsEMSDKNode>
```
Comment (radical) [inline src/libraries/sendtohelix-wasm.targets:49]: Why this change?

File: src/libraries/sendtohelixhelp.proj [hunk 2]
```diff
@@ -124,6 +124,11 @@
     <HelixPreCommand Include="printenv | grep COMPlus" />
   </ItemGroup>
 
+  <ItemGroup>
+    <HelixPreCommand Condition="'$(WindowsShell)' == 'true'" Include="set DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1" />
+    <HelixPreCommand Condition="'$(WindowsShell)' != 'true'" Include="export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1" />
+  </ItemGroup>
```
Comment (radical) [inline src/libraries/sendtohelixhelp.proj:130]: I think this shouldn't be needed because we set `DOTNET_CLI_HOME` to a writeable location now, its line 132-133.

File: src/libraries/sendtohelix-wasm.targets (hunk 1 above)
Comment (radekdoulik) [reply src/libraries/sendtohelix-wasm.targets:49]: It comes from https://github.com/dotnet/runtime/commit/37080d6d5d32de323498d60b11ee24e873e56a3b. We use the newer node from the docker image.

**Comment by radekdoulik** [comment]
The `runtime-wasm (Build browser-wasm linux Release LibraryTests_NodeJs)` fails on main too.

**Comment by lewing** [comment]
> The `runtime-wasm (Build browser-wasm linux Release LibraryTests_NodeJs)` fails on main too.

@maraf are you looking into this?

File: eng/testing/WasmRunnerAOTTemplate.sh [hunk 3]
```diff
@@ -35,8 +35,10 @@ if [[ "$XHARNESS_COMMAND" == "test" ]]; then
 	if [[ -z "$JS_ENGINE" ]]; then
 		if [[ "$SCENARIO" == "WasmTestOnNodeJS" || "$SCENARIO" == "wasmtestonnodejs" ]]; then
 			JS_ENGINE="--engine=NodeJS"
+			JS_ENGINE_ARGS="$JS_ENGINE_ARGS --engine-arg=--experimental-wasm-simd"
 		else
 			JS_ENGINE="--engine=V8"
+			JS_ENGINE_ARGS="$JS_ENGINE_ARGS --engine-arg=--experimental-wasm-simd"
```
Comment (kg) [inline eng/testing/WasmRunnerAOTTemplate.sh:41]: are we going to keep any lane that tests what happens if simd is disabled? or are we just going to require wasm simd for P3 and later, period. i don't have a good sense of how widespread browsers without wasm simd support are and whether we need to have automated test coverage for what happens

File: src/mono/wasm/templates/templates/console/runtimeconfig.template.json [hunk 4]
```diff
@@ -4,12 +4,10 @@
...
-                "Host": "nodejs"
-            },
-            {
-                "name": "v8",
-                "js-path": "main.mjs",
-                "Host": "v8"
+                "Host": "nodejs",
+                "host-args": [
+                    "--experimental-wasm-simd"
+                ]
```
Comment (kg) [inline src/mono/wasm/templates/templates/console/runtimeconfig.template.json:10]: why does this diff show v8 being removed?

**Review by kg — APPROVED** [review]
Looks good, with the caveat that I don't have the ability to fully vet the yml or msbuild bits (but the latter do look correct to me), and I left some comments

File: src/mono/wasm/templates/templates/console/runtimeconfig.template.json (hunk 4 above)
Comment (radekdoulik) [reply src/mono/wasm/templates/templates/console/runtimeconfig.template.json:10]: @radical is that intentional?

Comment (radical) [reply src/mono/wasm/templates/templates/console/runtimeconfig.template.json:10]: We don't support v8 for the console template, only nodejs.

Comment (radekdoulik) [reply src/mono/wasm/templates/templates/console/runtimeconfig.template.json:10]: so this is unrelated to SIMD, right?

File: eng/testing/WasmRunnerAOTTemplate.sh (hunk 3 above)
Comment (radekdoulik) [reply eng/testing/WasmRunnerAOTTemplate.sh:41]: I will add this test in folloup PR.

Comment (radekdoulik) [reply eng/testing/WasmRunnerAOTTemplate.sh:41]: https://github.com/dotnet/runtime/pull/82845
